code/modules/mob/mob_grab_specials.dm

```diff
@@ -47,19 +47,16 @@
 	if(state < GRAB_AGGRESSIVE)
 		to_chat(attacker, SPAN_WARNING("You require a better grab to do this."))
 		return
-
 	var/obj/item/organ/external/organ = target.get_organ(check_zone(target_zone))
 	if(!organ || organ.dislocated == -1)
 		return
 
-	var/time_to_jointlock = max( 0, ( target.getarmor(target_zone, ARMOR_MELEE) - attacker.stats.getStat(STAT_ROB) ) )
-	if(!do_mob(attacker, target, time_to_jointlock))
-		attacker << SPAN_WARNING("You must stand still to jointlock [target]!")
+	if(!do_after(attacker, 70, target))
```

Comment:
Flat seven seconds? That's quite long for a combat action.
```suggestion
	if(!do_after(attacker, 7 SECONDS, target))
```

Reply:
lotsa pain, intended, cool and good